File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
       Stack<Character> st=new Stack<>();
       for(char ch:s.toCharArray()) st.push(ch);

       for(int i=t.length()-1;i>=0;i--){
        if(st.isEmpty()) return true;
        if(t.charAt(i)==st.peek()) st.pop();
       }
        if(st.isEmpty()) return true;
        return false;
    }
}

File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        count=0
        if(len(s)==0):
            return True
        i=0
        for c in t:
            if(c==s[i]):
                i+=1
            if i==len(s):
                return True
        return False
        